### PR TITLE
Added postgres 14 to support versions(including breeze)

### LINKED ISSUE #23460: Add Postgres 14 support
### Description

_No response_

### Use case/motivation

Using Postgres 14 as backend

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by subkanthi**
closes: #23460

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by o-nikolas — APPROVED** [review]

File: scripts/ci/libraries/_initialization.sh [hunk 1]
```diff
@@ -125,7 +125,7 @@ function initialization::initialize_base_variables() {
     export CURRENT_PYTHON_MAJOR_MINOR_VERSIONS
 
     # Currently supported versions of Postgres
-    CURRENT_POSTGRES_VERSIONS+=("10" "13")
+    CURRENT_POSTGRES_VERSIONS+=("10" "13" "14")
```
Comment (potiuk) [inline scripts/ci/libraries/_initialization.sh]:
```suggestion
    CURRENT_POSTGRES_VERSIONS+=("10" "14")
```

Comment (potiuk) [reply scripts/ci/libraries/_initialization.sh]: I think we are perfectly ok with just runnning tests for "10" and "14" - postgres is stable enough to trust versions in-between are not broken.

**Review by potiuk — APPROVED** [review]
Looks cool. Can you also rebase and regenerate the images using latest rich /pre-commit - the screenshots are conflicting because of change in rich 12.4.0 that made the screenshots nicer but very different and it's been merged in #23570

File: scripts/ci/libraries/_initialization.sh (hunk 1 above)
Comment (ashb) [reply scripts/ci/libraries/_initialization.sh]: Was coming to suggest the same thing.

**Comment by potiuk** [comment]
Good point @eladkal !

I'd also be inclined to add Postgres 14 to supported in "2.3.0" and also in [rerequisites https://airflow.apache.org/docs/apache-airflow/stable/installation/prerequisites.html 

Since we have not made any change to get "14" supported, it's likely also no change for 2.3.0 - and we can easily cherry-pick that change and run tests in v2-3-test to run the tests on postgres 14 in the v2-3-test branch from now on.

**Comment by potiuk** [comment]
You want to rebase to latest version for rich 12.4.1 min-version after #23604  is merged.

**Comment by potiuk** [comment]
Yeah @subkanthi  -> I made the changes so that in the future it does not matter how old your breeze installation is. Please rebase to latest master (replace all the changes in images/breeze with those from main) and re-run pre-commit. It should work fine then.

**Comment by subkanthi** [comment]
> Waaaay better now :). My fixe seemed to work :)

@potiuk , sorry noticed I accidentally removed k8s version(1.24) in README.md, created PR

https://github.com/apache/airflow/pull/23649/files

**Comment by raphaelauv** [comment]
@potiuk release 2.3.0 was no officially tested again postgres 14

the README should not say so ->

Screenshot from 2022-05-11 18-38-13

**Comment by raphaelauv** [comment]
```
I believe we did not make any change to make it work except adding CI
```
yes

**Comment by potiuk** [comment]
Cool. Add PR to indicate it and I will cherry-pick it to 2.3 as well :D

**Pull request merged**